### PR TITLE
networkd: initcwn/initwnd use the right parsers

### LINKED ISSUE #7765: networkd: initcwn/initwnd doesn't use the right parsers
See my comments here:

https://github.com/systemd/systemd/commit/323d9329e7a3d62a76d2f60b6884e1aa69bfe685

/cc @ssahani

### PR DESCRIPTION
**Pull request by ssahani**
Closes #7765

### CODE REVIEW

File: src/network/networkd-route.c [hunk 1]
```diff
@@ -1107,10 +1107,10 @@ int config_parse_tcp_window(const char *unit,
         if (r < 0)
                 return r;
 
-        r = safe_atou32(rvalue, &k);
-        if (r < 0) {
+        r = parse_size(rvalue, 1024, &k);
```
Comment (yuwata) [inline src/network/networkd-route.c:1110]: Now `k` may be larger than `UINT32_MAX`. I think we should check the value is not larger than `UINT32_MAX`.

File: man/systemd.network.xml [hunk 2]
```diff
@@ -1070,19 +1070,19 @@
         <varlistentry>
           <term><varname>InitialCongestionWindow=</varname></term>
           <listitem>
-            <para>The TCP initial congestion window or <literal>InitialCongestionWindow</literal>  is used during the start
-            of a TCP connection. During the start of a TCP session, when a client requests for a resource, the server's initial congestion window
-            determines how many data packets will be sent during the initial burst of data. Takes a number between between 1 and 4294967295 (2^32 - 1).
+            <para>The TCP <literal>InitialCongestionWindow</literal> is used during the start of a TCP connection. During the start
+            of a TCP session, when a client requests for a resource, the server's initial congestion window determines how many data bytes
+            will be sent during the initial burst of data. Takes a number between between 1 and 4294967295 (2^32 - 1).
```
Comment (yuwata) [inline man/systemd.network.xml]: The last sentence should mention the suffix. E.g., "Takes a size in bytes between 1 and 4294967295 (2^32 - 1). The usual suffixes K, M, G are supported and are understood to the base of 1024."

File: man/systemd.network.xml [hunk 3]
```diff
@@ -1070,19 +1070,19 @@
...
         </varlistentry>
         <varlistentry>
           <term><varname>InitialAdvertisedReceiveWindow=</varname></term>
           <listitem>
-            <para>The TCP receive window size or <literal>InitialAdvertisedReceiveWindow</literal> is the amount of receive data (in bytes)
-            that can be buffered at one time on a connection. The sending host can send only that amount of data before waiting for
-            an acknowledgment and window update from the receiving host. Takes a number between 1 and 4294967295 (2^32 - 1). Defaults to unset.
+            <para>The TCP <literal>InitialAdvertisedReceiveWindow</literal> is the amount of receive data (in bytes) that can be buffered
+            at one time on a connection. The sending host can send only that amount of data before waiting for an acknowledgment and window
+            update from the receiving host. Takes a number between 1 and 4294967295 (2^32 - 1). Defaults to unset.
```
Comment (yuwata) [inline man/systemd.network.xml]: The same as the above.

File: man/systemd.network.xml (hunk 2 above)
Comment (yuwata) [reply man/systemd.network.xml]: Also, `0` means not using this feature, right? If so, I think it is worth to mention that.

Comment (ssahani) [reply man/systemd.network.xml]: No kernel just sets to TCP_INIT_CWND           10

File: man/systemd.network.xml [hunk 4]
```diff
@@ -1070,19 +1070,19 @@
         <varlistentry>
           <term><varname>InitialCongestionWindow=</varname></term>
           <listitem>
-            <para>The TCP initial congestion window or <literal>InitialCongestionWindow</literal>  is used during the start
-            of a TCP connection. During the start of a TCP session, when a client requests for a resource, the server's initial congestion window
-            determines how many data packets will be sent during the initial burst of data. Takes a number between between 1 and 4294967295 (2^32 - 1).
+            <para>The TCP <literal>InitialCongestionWindow</literal> is used during the start of a TCP connection. During the start
```
Comment (poettering) [inline man/systemd.network.xml]: why write "initial congestion window" as `<literal>InitialCongestionWindow</literal>`?

This "fix" is the wrong way round. We very much prefer the English language over some weird literal terms...

File: man/systemd.network.xml [hunk 5]
```diff
@@ -1070,19 +1070,19 @@
...
             </para>
           </listitem>
         </varlistentry>
         <varlistentry>
           <term><varname>InitialAdvertisedReceiveWindow=</varname></term>
           <listitem>
-            <para>The TCP receive window size or <literal>InitialAdvertisedReceiveWindow</literal> is the amount of receive data (in bytes)
-            that can be buffered at one time on a connection. The sending host can send only that amount of data before waiting for
-            an acknowledgment and window update from the receiving host. Takes a number between 1 and 4294967295 (2^32 - 1). Defaults to unset.
+            <para>The TCP <literal>InitialAdvertisedReceiveWindow</literal> is the amount of receive data (in bytes) that can be buffered
```
Comment (poettering) [inline man/systemd.network.xml]: as above

File: man/systemd.network.xml [hunk 6]
```diff
@@ -1070,19 +1070,19 @@
...
+            at one time on a connection. The sending host can send only that amount of data before waiting for an acknowledgment and window
+            update from the receiving host. Takes a number between 1 and 4294967295 (2^32 - 1). Defaults to unset.
             </para>
```
Comment (poettering) [inline man/systemd.network.xml]: also, should briefly mention that the "K", "M", "G" suffixes are supported too

**Comment by ssahani** [comment]
updated.

File: man/systemd.network.xml [hunk 7]
```diff
@@ -1070,19 +1070,20 @@
         <varlistentry>
           <term><varname>InitialCongestionWindow=</varname></term>
           <listitem>
-            <para>The TCP initial congestion window or <literal>InitialCongestionWindow</literal>  is used during the start
-            of a TCP connection. During the start of a TCP session, when a client requests for a resource, the server's initial congestion window
-            determines how many data packets will be sent during the initial burst of data. Takes a number between between 1 and 4294967295 (2^32 - 1).
-            Defaults to unset.
+            <para>The TCP initialcongestion window is used during the start of a TCP connection. During the start of a TCP
```
Comment (yuwata) [inline man/systemd.network.xml]: 'initial congestion': missing space.

**Comment by ssahani** [comment]
updated

File: man/systemd.network.xml [hunk 8]
```diff
@@ -1070,19 +1070,20 @@
         <varlistentry>
           <term><varname>InitialCongestionWindow=</varname></term>
           <listitem>
-            <para>The TCP initial congestion window or <literal>InitialCongestionWindow</literal>  is used during the start
-            of a TCP connection. During the start of a TCP session, when a client requests for a resource, the server's initial congestion window
-            determines how many data packets will be sent during the initial burst of data. Takes a number between between 1 and 4294967295 (2^32 - 1).
-            Defaults to unset.
+            <para>The TCP initial congestion window is used during the start of a TCP connection. During the start of a TCP
+            session, when a client requests for a resource, the server's initial congestion window determines how many data bytes
```
Comment (keszybz) [inline man/systemd.network.xml]: "requests a resource"

"will be sent *in* the initial burst of data"

File: man/systemd.network.xml [hunk 9]
```diff
@@ -1070,19 +1070,20 @@
...
             </para>
           </listitem>
         </varlistentry>
         <varlistentry>
           <term><varname>InitialAdvertisedReceiveWindow=</varname></term>
           <listitem>
-            <para>The TCP receive window size or <literal>InitialAdvertisedReceiveWindow</literal> is the amount of receive data (in bytes)
-            that can be buffered at one time on a connection. The sending host can send only that amount of data before waiting for
-            an acknowledgment and window update from the receiving host. Takes a number between 1 and 4294967295 (2^32 - 1). Defaults to unset.
+            <para>The TCP initial advertised receive window is the amount of receive data (in bytes) that can be buffered at one time
```
Comment (keszybz) [inline man/systemd.network.xml]: "can be buffered" → "can initially be buffered"

**Comment by keszybz** [comment]
C'mon, "in during" ?!

**Comment by ssahani** [comment]
did not get .

**Comment by poettering** [comment]
> did not get .

As @keszybz points out "in during" is not proper English. Please correct it. Thanks.

**Pull request merged**